Teuchos:  clang reporting improper cast in Teuchos_LAPACK.cpp

## Bug Report
@bartlettroscoe not really sure who owns this one but the error is happening when compiling _Teuchos\_LAPACK.cpp_ so you're the "lucky" winner.

### Description
As of last night I'm getting the following error when compiling _Teuchos\_LAPACK.cpp_ from the _develop_ branch:
```
/Trilinos/packages/teuchos/numerics/src/Teuchos_LAPACK.cpp:2096:49: error: implicit conversion from '_Complex double' to 'const double' is not permitted in C++
    return static_cast<Kokkos::complex<double>>(z);
           ~~~~~~~~~~~                          ^
1 error generated.
```

--- a/packages/teuchos/numerics/src/Teuchos_LAPACK.cpp
+++ b/packages/teuchos/numerics/src/Teuchos_LAPACK.cpp
@@ -2093,7 +2093,7 @@ namespace Teuchos
   Kokkos::complex<double> LAPACK<int, Kokkos::complex<double> >::LARND( const int& idist, int* seed ) const
   {
     double _Complex z = ZLARND_F77(&idist, seed);
-    return static_cast<Kokkos::complex<double>>(z);
+    return reinterpret_cast<Kokkos::complex<double>&>(z);
   }
 #endif
 

